### ISSUE TITLE
How can I use autocomplete?

### ISSUE THREAD

**Issue by Favianl** [AstroNvim/AstroNvim#645]
Hi, I'm a new neovim user. excuse me my english is basic.
I installed AstroNvim and then I also installed: html, css, emmet, and tsserver languages, using the commands :LspInstall and :TSInstall.

I would like to know how to make it show me suggestions and autocomplete when writing the code, autocomplete and hint works fine in html and css, but in javascript it doesn't.

I don't know what I need to configure, this is what it shows me in command :LspInfo :

![html](https://user-images.githubusercontent.com/106279874/173738992-8cd80b52-b7e4-46d9-aef7-b652e97503cc.jpg)
![css](https://user-images.githubusercontent.com/106279874/173739012-a7e54e37-0ddc-4c5a-82c3-97f644baf184.jpg)
![error-tsserver](https://user-images.githubusercontent.com/106279874/173739027-27a000fa-4c75-4409-8ae6-9c32258b9ee6.jpg)

I think the problem is that "not found" error which I don't know how to solve, hopefully you can help me.

**Comment by Johnsiras** [comment]
Do you have typescript-language-server install globally?

**Comment by Favianl** [comment]
> 



>

**Comment by Favianl** [comment]
> Do you have typescript-language-server installed globally?

I thought just :LspInstall tsserver was enough, but right now I just used :

npm install -g typescript-language-server typescript

still the same, same error...

**Comment by kabinspace** [comment]
@Favianl Since tsserver requires a config file, make a file called `tsconfig.json` in your root directory and put the given config in that file
```json
{
  "compilerOptions": {
    "module": "commonjs",
    "target": "es6",
    "checkJs": false
  },
  "exclude": [
    "node_modules"
  ]
}
```
I would also suggest you to check [docs](https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md#tsserver) for more information

**Comment by Johnsiras** [comment]
> @Favianl Since tsserver requires a config file, make a file called `tsconfig.json` in your root directory and put the given config in that file
> 
> ```json
> {
>   "compilerOptions": {
>     "module": "commonjs",
>     "target": "es6",
>     "checkJs": false
>   },
>   "exclude": [
>     "node_modules"
>   ]
> }
> ```
> 
> I would also suggest you to check [docs](https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md#tsserver) for more information

it's javascript tho not typescript?

**Comment by mehalter** [comment]
`tsserver` handles LSP for both typescript and JavaScript. It doesn't need to be a `tsconfig.json` file since that wouldn't make sense for a JavaScript project. It also looks for a `package.json` or `jsconfig.json` file in the root or even just a `.git` folder.

**Comment by Favianl** [comment]
> @Favianl Since tsserver requires a config file, make a file called `tsconfig.json` in your root directory and put the given config in that file
> 
> ```json
> {
>   "compilerOptions": {
>     "module": "commonjs",
>     "target": "es6",
>     "checkJs": false
>   },
>   "exclude": [
>     "node_modules"
>   ]
> }
> ```
> 
> I would also suggest you to check [docs](https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md#tsserver) for more information


Thanks for replying, I'm a bit confused where the tsconfig.json file should be placed, do you mean: /home/user ?

**Comment by mehalter** [comment]
@Favianl this should go in the root folder of your javascript project. These configuration files are on a per-project basis.

**Comment by Favianl** [comment]
> `tsserver` handles LSP for both typescript and JavaScript. It doesn't need to be a `tsconfig.json` file since that wouldn't make sense for a JavaScript project. It also looks for a `package.json` or `jsconfig.json` file in the root or even just a `.git` folder.

does it mean that if the file is only inside that project, creating a js file outside of it will give me the same error again?

**Comment by mehalter** [comment]
Yes, this is how Neovim checks if an LSP should start automatically. If you are not inside of a project you can run `:LspStart` to force it to start manually. But the automatic starting will only happen in recognizable javascript projects.

**Comment by Favianl** [comment]
> Yes, this is how Neovim checks if an LSP should start automatically. If you are not inside of a project you can run `:LspStart` to force it to start manually. But the automatic starting will only happen in recognizable javascript projects.

ok thanks for your help, this is working.
